db/hybrid_logical_vector.go

```diff
@@ -30,6 +31,85 @@ type Version struct {
 	Value uint64 `json:"version"`
 }
 
+// VersionsDeltas will be sorted by version, first entry will be fill version then after that will be calculated deltas
+type VersionsDeltas []Version
+
+func (vde VersionsDeltas) Len() int { return len(vde) }
+
+func (vde VersionsDeltas) Swap(i, j int) {
+	vde[i], vde[j] = vde[j], vde[i]
+}
+
+func (vde VersionsDeltas) Less(i, j int) bool {
+	if vde[i].Value == vde[j].Value {
+		return vde[i].SourceID < vde[j].SourceID
+	}
+	return vde[i].Value < vde[j].Value
+}
+
+// VersionDeltas calculate the deltas of input map
+func VersionDeltas(versions map[string]uint64) VersionsDeltas {
+	if versions == nil {
+		return nil
+	}
+
+	vdm := make(VersionsDeltas, 0, len(versions))
+	for src, vrs := range versions {
+		vdm = append(vdm, CreateVersion(src, vrs))
+	}
+
+	// sort the list
+	sort.Sort(vdm)
+
+	// traverse in reverse order and calculate delta between versions, leaving the first element as is
+	for i := len(vdm) - 1; i >= 1; i-- {
+		vdm[i].Value = vdm[i].Value - vdm[i-1].Value
+	}
+	return vdm
+}
+
+// VersionsToDeltas will calculate deltas from the input map (pv or mv). Then will return the deltas in persisted format
+func VersionsToDeltas(m map[string]uint64) []string {
+	if len(m) == 0 {
+		return nil
+	}
+
+	var vrsList []string
+	deltas := VersionDeltas(m)
+	for _, delta := range deltas {
+		key := delta.SourceID
+		val := delta.Value
+		encodedVal := base.Uint64ToLittleEndianHexAndStripZeros(val)
+		listItem := encodedVal + "@" + key
```

Comment:
It's a bit subtle that this format is different than delta.String().  I see now that these are two different representations of the version (simple hex used by blip, LittleEndian with strippedzeros here), but I think that's easy to miss.  What do you think about a Version.StringForVersionDelta() function with a comment explaining usage, that does lines 82 and 83?